### DISCUSSION TITLE
[React Native]: useQuery returned data don't have undefined as a type

### DISCUSSION THREAD

**Discussion by rajatabcx** [TanStack/query#6071] [Q&A] [answered]
Hi, I am using `@tanstack/react-query v4.29.15`, in both my web and mobile app, the webapp works fine where in useQuery, the returned data has type of `SpecifiedType|undefined` but in mobile app the data has type of `SpecifiedType` as we can see in the image, the app is in `expo react-native`, the undefined thing is right and it eases things out for conditional rendering but in mobile app it creates problem as I am using typescript

Is it a natural behavious, is it suppose to work like this or am i missing something

Hook Usage 
```js 
const { data, isLoading: loadingPlanData } = usePlan();
```
Hook
```js
export const usePlan = (enabled = true) => {
  const axiosPrivate = useAxiosPrivate();
  return useQuery(
    ['plan'],
    async () => {
      const res = await plan(axiosPrivate);
      return res.data;
    },
    {
      enabled,
    }
  );
};
```
Web App:
![image](https://github.com/TanStack/query/assets/108508250/b323b8f4-aee8-4fc6-8e58-89229f220901)
Mobile App:
![image](https://github.com/TanStack/query/assets/108508250/8023a5e2-e8ad-40d1-9c96-ed441dd3c565)

**Comment by TkDodo** [comment, marked as answer]
do both apps have the same `tsconfig` settings ?

**Reply by rajatabcx** [reply]
I just checked, they weren't same, I added `strict:true` in the config and now it is perfectly fine

Thanks!!
